exhale/graph.py

```diff
@@ -1116,6 +1152,75 @@ def discoverAllNodes(self):
+                    if title and title.string:
+                        page.title = title.string
+
+                    err_non = "[CRITICAL] did not find refid [{0}] in `self.node_by_refid`."
+                    err_dup = "Conflicting page definition: [{0}] appears to be defined in both [{1}] and [{2}]."  # noqa
+                    # process subpages
+                    inner_pages = cdef.find_all("innerpage", recursive=False)
+
+                    utils.verbose_log(
+                        "*** [{0}] had [{1}] innerpages found".format(page.name, len(inner_pages)),
+                        utils.AnsiColors.BOLD_MAGENTA
+                    )
+
+                    for subpage in inner_pages:
+                        if "refid" in subpage.attrs:
+                            refid = subpage.attrs["refid"]
+                            if refid in self.node_by_refid:
+                                node = self.node_by_refid[refid]
+
+                                # << verboseBuild
+                                utils.verbose_log(
+                                    "    - [{0}]".format(node.name),
+                                    utils.AnsiColors.BOLD_MAGENTA
+                                )
+
+                                if node.parent:
+                                    utils.verbose_log(
+                                        err_dup.format(node.name, node.parent.name, page.name),
+                                        utils.AnsiColors.BOLD_YELLOW
+                                    )
+
+                                if node not in page.children:
+                                    page.children.append(node)
+                                    node.parent = page
+                            else:
+                                # << verboseBuild
+                                utils.verbose_log(err_non.format(refid), utils.AnsiColors.BOLD_RED)
+
+                    # the location of the page as determined by doxygen
+                    location = cdef.find("location")
+                    if location and "file" in location.attrs:
+                        location_str = os.path.normpath(location.attrs["file"])
+                        # some older versions of doxygen don't reliably strip from path
+                        # so make sure to remove it
+                        abs_strip_path = os.path.normpath(os.path.abspath(
+                            configs.doxygenStripFromPath
+                        ))
+                        if location_str.startswith(abs_strip_path):
+                            location_str = os.path.relpath(location_str, abs_strip_path)
+                        page.location = os.path.normpath(location_str)
+
+                except:
+                    utils.fancyError(
+                        "Could not process Doxygen xml for file [{0}]".format(f.name)
+                    )
+        self.pages = [page for page in self.pages if not page.parent]
```

Comment:
Ok so basically you avoid needing to reparent by just setting the children as you set this up, and we can rely on innerpage definitions being present?  Nice

Reply:
Exactly. Checking Doxygen generated XML, I believe this holds.